editor/src/components/inspector/fill-hug-fixed-control.tsx

```diff
@@ -375,68 +371,279 @@ export const FillHugFixedControl = React.memo<FillHugFixedControlProps>((props)
+    dimensionControlStatus: ControlStatus
+    dimensionNumberType: CSSNumberType
+    fixedHugSelectOption: SelectOption | null
+    fixedHugControlStyles: ControlStyles
+    onAdjustDimension: OnSubmitValueOrUnknownOrEmpty<CSSNumber>
+    onSubmitFixedHug: (option: SelectOption) => void
+    isGroupChild: boolean
+    groupChildConstraint: GroupChildConstraintOptionType | 'mixed'
+  }) => {
+    const fixedHugFillOptions = useEditorState(
+      Substores.metadata,
+      fixedHugFillOptionsSelector,
+      'DimensionRow options',
+    )
+
+    const gridTemplate = React.useMemo((): GridRowVariant => {
+      return isGroupChild ? '|--67px--|<--------1fr-------->' : '<--1fr--><--1fr-->'
+    }, [isGroupChild])
+
+    const labelInner = React.useMemo(() => {
+      return dimension.charAt(0).toUpperCase()
+    }, [dimension])
+
+    return (
+      <UIGridRow padded={false} variant={gridTemplate} css={InspectorRowHoverCSS}>
         <NumberInput
-          id={FillFixedHugControlId('height')}
-          testId={FillFixedHugControlId('height')}
-          value={heightValue}
-          onSubmitValue={onAdjustHeight}
-          onTransientSubmitValue={onAdjustHeight}
-          onForcedSubmitValue={onAdjustHeight}
-          controlStatus={heightInputControlStatus}
-          numberType={pickNumberType(heightCurrentValue.fixedHugFill)}
+          labelInner={labelInner}
+          id={FillFixedHugControlId(dimension)}
+          testId={FillFixedHugControlId(dimension)}
+          value={dimensionValue}
+          onSubmitValue={onAdjustDimension}
+          onTransientSubmitValue={onAdjustDimension}
+          onForcedSubmitValue={onAdjustDimension}
+          controlStatus={dimensionControlStatus}
+          numberType={dimensionNumberType}
           incrementControls={true}
           stepSize={1}
           minimum={0}
           maximum={Infinity}
           defaultUnitToHide={null}
           focusOnMount={false}
         />
+        {isGroupChild ? (
+          <GroupConstraintSelect dimension={dimension} type={groupChildConstraint} />
+        ) : (
+          <PopupList
+            value={fixedHugSelectOption ?? undefined}
+            options={fixedHugFillOptions}
+            onSubmitValue={onSubmitFixedHug}
+            controlStyles={fixedHugControlStyles}
+          />
+        )}
       </UIGridRow>
-    </FlexCol>
-  )
-})
+    )
+  },
+)
+
+DimensionRow.displayName = 'DimensionRow'
+
+const GroupConstraintSelect = React.memo(
+  ({
+    dimension,
+    type,
+  }: {
+    dimension: LayoutPinnedProp
+    type: GroupChildConstraintOptionType | 'mixed'
+  }) => {
+    const dispatch = useDispatch()
+    const colorTheme = useColorTheme()
+
+    const editorRef = useRefEditorState((store) => ({
+      selectedViews: store.editor.selectedViews,
+      allElementProps: store.editor.allElementProps,
+    }))
+
+    function optionFromType(value: GroupChildConstraintOptionType | 'mixed') {
+      return value === 'constrained'
```

Comment:
Since this is already in a function, I feel like this would be clearer just using regular if..else statements.

Reply:
done in [`167c79a` (#4159)](https://github.com/concrete-utopia/utopia/pull/4159/commits/167c79af2bf4960d669631f31064c0ccbf6199de)